typewritten.zsh

```diff
@@ -47,6 +47,8 @@ tw_redraw() {
   local tw_virtual_env=""
   if [[ ! -z $VIRTUAL_ENV ]] && [[ -z $VIRTUAL_ENV_DISABLE_PROMPT ]]; then
     tw_virtual_env="%F{$tw_colors[virtual_env]}($(basename $VIRTUAL_ENV)) "
+  elif [[ ! -z $CONDA_PREFIX ]] && [[ -z $VIRTUAL_ENV_DISABLE_PROMPT ]]; then
+    tw_virtual_env="%F{$tw_colors[virtual_env]}($(basename $CONDA_PREFIX)) "
```

Comment:
I have been wanting to have a better separation of concerns within the prompt for a while.

Since we are working on this part, let's use the time to do it.

I'm thinking of a function `tw_get_virtual_env` that can be called to define `tw_env_prompt`.

```bash
tw_get_virtual_env() {
  if [[ -z $VIRTUAL_ENV_DISABLE_PROMPT ]]; then
    local tw_virtual_env=""
    if [[ ! -z $VIRTUAL_ENV ]]; then
      tw_virtual_env="$VIRTUAL_ENV"
    elif [[ ! -z $CONDA_PREFIX ]]; then
      tw_virtual_env="$CONDA_PREFIX"
    fi;

    if [[ $tw_virtual_env != "" ]]; then
      echo "%F{$tw_colors[virtual_env]}($(basename $tw_virtual_env)) "
    fi;
  fi;
}
```

The function can be right at the top of `tw_redraw`.

Then, the `tw_env_prompt` can be defined like so:
```bash
tw_env_prompt="$(tw_get_virtual_env)$tw_prompt"
```

What do you think? It's a bit less clean than what is currently there, but the logic is separated and is less intimidating to work in I think.

Feedback welcome! Thanks again for your work!

Reply:
BTW I didn't test my function extensively so we need some testing to make sure it's ok when we come up with a final version.

Reply:
Yah this sounds like the better way to do it.  Do you mean having `tw_get_virtual_env` nested within `tw_redraw`, or have it sequentially on top of `tw_redraw`?  I think you probably mean the latter, but I'm not much of an expert with zsh.

Reply:
The latter sound good!